src/v/cluster/archival_metadata_stm.cc

```diff
@@ -620,9 +620,145 @@ ss::future<std::error_code> archival_metadata_stm::process_anomalies(
     co_return co_await builder.replicate();
 }
 
+ss::future<bool>
+archival_metadata_stm::sync(model::timeout_clock::duration timeout) {
```

Comment:
Note for other reviewers: this method is called from the upload loop before any decisions are made. We want to make sure that the decisions are based on up to date state.